Backpack/Dialog/Classes/BPKDialogController.m

```diff
@@ -0,0 +1,224 @@
+                                iconImage:(UIImage *)iconImage {
+    return [[self alloc] initWithTitle:title message:message style:style shadowStyle:shadowStyle headColor:headColor iconImage:iconImage];
+}
+
+- (void)setupViews {
+    UITapGestureRecognizer *tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(scrimTapped:)];
+    
+    self.scrimView = [UIView new];
+    self.scrimView.clipsToBounds = YES;
+    self.scrimView.userInteractionEnabled = YES;
+    self.scrimView.backgroundColor = BPKColor.gray900;
+    self.scrimView.alpha = 0.5;
+    self.scrimView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self.scrimView addGestureRecognizer:tapGesture];
+
+    self.dialogView = [[BPKDialogView alloc] initWithFrame:CGRectZero];
+    self.dialogView.translatesAutoresizingMaskIntoConstraints = NO;
+    self.dialogView.delegate = self;
+    [self.dialogView setTitle:self.titleText];
+    [self.dialogView setIconImage:self.iconImage];
+    self.dialogView.hasShadow = self.shadowStyle == BPKDialogControllerShadowStyleShadow;
+    [self.dialogView setDescription:self.messageText];
+    [self.dialogView setIconBackgroundColor:self.headColor];
+}
+
+- (void)setupConstraints {
+    [NSLayoutConstraint activateConstraints:@[
+                                              [self.scrimView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
+                                              [self.scrimView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
+                                              [self.scrimView.topAnchor constraintEqualToAnchor:self.view.topAnchor],
+                                              [self.scrimView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor],
+
+                                              [self.dialogView.leadingAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.leadingAnchor constant:(BPKSpacingMd + BPKSpacingSm)],
+                                              [self.view.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.dialogView.trailingAnchor constant:(BPKSpacingMd + BPKSpacingSm)],
+                                              ]];
+    
+    [self.dialogView.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor].active = YES;
+    if (self.style == BPKDialogControllerStyleBottomSheet) {
+        self.bottomAnchorConstraint = [self.dialogView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor];
+        self.bottomAnchorConstraint.active = YES;
+    } else if (self.style == BPKDialogControllerStyleAlert) {
+        [self.dialogView.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor].active = YES;
+    }
+}
+
+- (void)addViews {
+    [self.view addSubview:self.scrimView];
+    [self.view addSubview:self.dialogView];
+}
+
+- (void)scrimTapped:(UITapGestureRecognizer *)gestureRecognizer {
+    if( gestureRecognizer.state == UIGestureRecognizerStateRecognized ) {
```

Comment:
inconsistent coding style `if( xxx )` vs `if (xxx)`everywhere else

Reply:
Good catch, should be tight